File: src/main/java/com/dayo/testgame/G.java
package com.dayo.testgame;

import com.dayo.simplegameapi.api.Game;
import com.dayo.simplegameapi.data.RoomInfo;
import com.dayo.simplegameapi.event.PlayerFailEvent;

import java.util.List;
import java.util.UUID;

public class G extends Game {
    @Override
    public int getId() {
        return 0;
    }

    @Override
    public int getMinimumPlayer() {
        return 1;
    }

    @Override
    public String getName() {
        return "asdf";
    }

    @Override
    public void onGameStart(RoomInfo roomInfo, List<UUID> list) {
        TestGame.instance.getServer().getPlayer(list.get(0)).sendMessage("Hello, world!");
    }
}

File: src/main/java/com/dayo/testgame/G.kt
package com.dayo.testgame

import me.ddayo.coroutine.Coroutine
import me.ddayo.coroutine.functions.WaitNextTick
import me.ddayo.simplegameapi.api.Game
import me.ddayo.simplegameapi.data.GameManager
import me.ddayo.simplegameapi.data.RoomInfo
import me.ddayo.simplegameapi.util.CoroutineUtil
import org.bukkit.Bukkit
import org.bukkit.scheduler.BukkitTask
import java.util.*

class G : Game {
    private lateinit var task: BukkitTask
    override val playerCount: Int
        get() = 1
    override val name: String
        get() = "asdf"
    override val maxPlayerCount: Int
        get() = 2

    lateinit var p: UUID

    constructor(a: Int): super() {
        println(a)
    }

    constructor(): super()

    override fun onGameStart(players: List<UUID>) {
        //TestGame.instance!!.server.getPlayer(players[0])!!.sendMessage("Hello, world!")
        players.forEach { TestGame.instance!!.server.getPlayer(it)!!.sendMessage(room.rid.toString()) }
        println("t")
        println(GameManager.getPlaying(players[0])?.rid)
        println(players[0])
        p = players[0]
        Coroutine.startCoroutine(sequence {
            while(true) {
                if (TestGame.stop) {
                    TestGame.stop = false
                    TestGame.instance!!.server.getPlayer(p)!!.sendMessage("Stop game!")
                    playerFailed(p)
                    return@sequence
                }
                yield(WaitNextTick())
            }
        })
    }

    override fun playerFailed(player: UUID) {
        TestGame.instance!!.server.getPlayer(player)!!.sendMessage("You Failed")
        super.playerFailed(player)
    }

    override fun finish() {
        TestGame.instance!!.server.broadcastMessage("Finish game")
        super.finish()
        CoroutineUtil.invokeMain {
            println(GameManager.getRoomStatus(room))
        }
    }
}